src/Block/GalleryBlockService.php

```diff
@@ -196,6 +201,33 @@ public function buildEditForm(FormMapper $formMapper, BlockInterface $block)
         ]);
     }
 
+    /**
+     * NEXT_MAJOR: delete it.
+     */
+    public function buildCreateForm(FormMapper $formMapper, BlockInterface $block)
+    {
+        $this->configureCreateForm($formMapper, $block);
```

Comment:
IMO, we should add deprecation messages for methods which will be removed in the next major release. I'd use a similar approach to I've done [here](https://github.com/sonata-project/SonataAdminBundle/blob/c22455c61e585496cb01a94021fa96173704de4c/src/Admin/AbstractAdmin.php#L1593-L1598) in order to avoid our own code to trigger the deprecation.

Reply:
Something like this?
```
public function buildCreateForm(FormMapper $formMapper, BlockInterface $block)
{
        if ('sonata_deprecation_mute' !== (\func_get_args()[0] ?? null)) {
            @trigger_error(sprintf(
                'Using this method %s() is deprecated since sonata-project/admin-bundle 3.x. It will be removed in version 4.0.',
                __METHOD__
            ), E_USER_DEPRECATED);
        }
```

Reply:
Yes, in this case:
```php
    public function buildCreateForm(FormMapper $formMapper, BlockInterface $block)
    {
        if ('sonata_deprecation_mute' !== (\func_get_args()[2] ?? null)) {
            @trigger_error(sprintf(
                'Method %s() is deprecated since sonata-project/media-bundle 3.x and will be removed in version 4.0. Use %s::configureCreateForm() instead.',
                __METHOD__,
                static::class,
            ), E_USER_DEPRECATED);
        }
    }
```

Reply:
Are these right now?